### PR TITLE
Send artifact events

### PR DESCRIPTION
**Pull request by t-persson**
### Applicable Issues

fixes: eiffel-community/etos#15

### Description of the Change

Move log directory handling to workspace
Create new log area library for collecting and uploading logs and
artifacts
Disable override of log and artifact directories
Separate artifacts and logs
Send artifact events for artifacts

### Alternate Designs

We could send a single ArtifactCreated for each artifact created instead of grouping them with the test case.
This would increase granularity if there are a lot of artifacts created. However i believe that grouping is nicer and easier for consumers.
One could also group even more and only send a single artifact created at the end of the test suite. However in that case we would not get artifacts in real-time.

### Benefits

Artifacts should be artifacts and logs should be logs.
Easier for ETOS Client to separate logs and artifacs.
The TestSuiteFinished event won't be as large.

### Possible Drawbacks

More than one place to look for "logs" when executing ETOS.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Tobias Persson <tobias.persson@axis.com>

### CODE REVIEW

**Review by andjoe-axis — APPROVED** [review]
+1

File: src/etos_test_runner/lib/log_area.py [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+            item = self._fix_name(item, test_name)
+            items.append({"name": item.name, "file": item})
+        return items
+
+    def upload_logs(self, logs):
+        """Upload logs to log area.
+
+        :param logs: Logs to upload.
+        :type logs: list
+        :return: Log name and URI
```
Comment (fredjn) [inline src/etos_test_runner/lib/log_area.py:135]: remove ?

File: src/etos_test_runner/lib/log_area.py [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+            artifact_created,
+            links={"CONTEXT": self.etos.config.get("context")},
+        )
+
+    def upload_artifacts(self, artifacts):
+        """Upload artifacts to log area.
+
+        :param artifacts: Artifacs to upload.
+        :type artifacts: list
+        :return: Artifact name and URI
```
Comment (fredjn) [inline src/etos_test_runner/lib/log_area.py:197]: remove ?

File: src/etos_test_runner/lib/log_area.py [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+        :type url: str
+        :param log_file: Opened log file to upload.
+        :type log_file: file
+        :param timeout: How long, in seconds, to retry request.
+        :type timeout: int or None
+        :param as_json: Whether or not to return json instead of response.
+        :type as_json: bool
+        :param request_kwargs: Keyword arguments for the requests command.
+        :type request_kwargs: dict
+        :return: HTTP response or json.
```
Comment (fredjn) [inline src/etos_test_runner/lib/log_area.py:289]: remove ?

**Review by fredjn — CHANGES_REQUESTED** [review]

File: src/etos_test_runner/lib/log_area.py (hunk 3 above)
Comment (t-persson) [reply src/etos_test_runner/lib/log_area.py:289]: This one actually returns

Comment (fredjn) [reply src/etos_test_runner/lib/log_area.py:289]: Ah true, perhaps an update to the return type?

**Review by fredjn — APPROVED** [review]

File: src/etos_test_runner/lib/log_area.py (hunk 3 above)
Comment (t-persson) [reply src/etos_test_runner/lib/log_area.py:285]: Nope. It's correct.

**Pull request merged**